axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/GrpcBackedCommandMessage.java

```diff
@@ -16,98 +16,85 @@
 
 package org.axonframework.axonserver.connector.command;
 
-import io.axoniq.axonserver.grpc.MetaDataValue;
 import io.axoniq.axonserver.grpc.command.Command;
-import org.axonframework.axonserver.connector.util.GrpcMetaDataConverter;
+import org.axonframework.axonserver.connector.util.GrpcMetaData;
+import org.axonframework.axonserver.connector.util.GrpcSerializedObject;
 import org.axonframework.commandhandling.CommandMessage;
 import org.axonframework.messaging.MetaData;
-import org.axonframework.serialization.SerializedObject;
+import org.axonframework.serialization.LazyDeserializingObject;
 import org.axonframework.serialization.Serializer;
-import org.axonframework.serialization.SimpleSerializedObject;
 
 import java.util.Map;
+import java.util.function.Supplier;
 
 /**
  * Wrapper that allows clients to access a gRPC {@link Command} as a {@link CommandMessage}.
  *
+ * @param <C> a generic specifying the payload type of the wrapped {@link CommandMessage}
  * @author Marc Gathier
  * @since 4.0
  */
 public class GrpcBackedCommandMessage<C> implements CommandMessage<C> {
 
-    private final Command request;
-    private final Serializer serializer;
-    private MetaData metaData;
+    private final Command command;
+    private final LazyDeserializingObject<C> serializedPayload;
+    private final Supplier<MetaData> metaDataSupplier;
 
     /**
      * Instantiate a {@link GrpcBackedCommandMessage} with the given {@code command} and using the provided
      * {@link Serializer} to be able to retrieve the payload and {@link MetaData} from it.
      *
      * @param command    the {@link Command} which is being wrapped as a {@link CommandMessage}
      * @param serializer the {@link Serializer} used to deserialize the payload and {@link MetaData} from the given
-     * {@code command}
+     *                   {@code command}
      */
     public GrpcBackedCommandMessage(Command command, Serializer serializer) {
-        this.request = command;
-        this.serializer = serializer;
+        this.command = command;
```

Comment:
You could invoke the `GrpcBackedCommandMessage(Command, LazyDeserializingObject, Supplier)` constructor.

Reply:
Good suggestions :+1: